### DISCUSSION TITLE
Regular expressions in paths

### DISCUSSION THREAD

**Discussion by rmwiseman** [bluenviron/mediamtx#619] [Questions]
I know that in v0.10.0, regular expressions were allowed in the list of paths.

The only documentation I can find on this is on #54 - the functionality is not mentioned in the README.md file.

Once the path has been deemed to match the regular expression, is that it?  At that point, does the fact that there was a regexp match get "thrown away" and the matching stream is returned?

I ask because that's what it seems to be (from what I can work out), but that doesn't quite match the functionality I would like.

I would like to have a path matching `/<label>/<UUID>` (so a regexp like `^/[a-z]+/([0-9-A-Fa-f-]+)`, to seriously over-simplify things) and then take whatever UUID was passed (`$1`) and use that in the `ffmpeg` command that matches the path.

So two questions:

1. Are capture groups (by which I mean parenthesised parts of the regexp) available to use in the command that matches the path?
2. Is a new command spawned for each different path that matches the regexp?

I suspect the answer is no to both of these, but before I add an enhancement request, I thought I would ask!

Thanks!

**Comment by aler9** [comment]
> Is a new command spawned for each different path that matches the regexp?

If for command you mean runOnDemand / runOnPublish / runOnRead, then yes, a new command is spawned for each path that matches the regexp.

> Are capture groups (by which I mean parenthesised parts of the regexp) available to use in the command that matches the path?

No, they are not, but this could be a good improvement. Open a feature request or provide a patch.

Documentation about regexp is fully available in the configuration file:

https://github.com/aler9/rtsp-simple-server/blob/main/rtsp-simple-server.yml#L119

**Reply by rmwiseman** [reply]
Thanks, I'll open a feature request.  And thanks for the clarification.

[a comment by an automated account is not shown]
